### PR TITLE
Refactor non-oss file loading

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

**Comment by jakolehm** [comment]
@kke PTAL

File: pharos-cluster.gemspec [hunk 1]
```diff
@@ -3,6 +3,13 @@ lib = File.expand_path("../lib", __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require "pharos/version"
 
+files = Dir['README', 'LICENSE', 'licenses/*', 'bin/*', 'lib/**/*', 'addons/**/*']
```
Comment (kke) [inline pharos-cluster.gemspec]:
```suggestion
files = Dir['README.md', 'LICENSE', 'licenses/*', 'bin/*', 'lib/**/*', 'addons/**/*']
```

File: pharos-cluster.gemspec [hunk 2]
```diff
@@ -3,6 +3,13 @@ lib = File.expand_path("../lib", __FILE__)
...
+require_paths = ['lib']
+if ENV['PHAROS_NON_OSS'].to_s == 'true'
```
Comment (kke) [inline pharos-cluster.gemspec]:
```suggestion
if ENV['PHAROS_NON_OSS'] == 'true'
```

`nil == anything` is false

**Comment by jakolehm** [comment]
@kke PTAL (part II)

File: lib/pharos_cluster.rb [hunk 3]
```diff
@@ -16,3 +16,8 @@ module Pharos
   COREDNS_VERSION = '1.1.3'
   TELEMETRY_VERSION = '0.1.0'
 end
+
+begin
+  require "pharos_non_oss"
+rescue LoadError # rubocop:disable Lint/HandleExceptions
```
[the remark this answers is not in the document]
Comment (jakolehm) [reply lib/pharos_cluster.rb]: Fixed.

**Pull request merged**